delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViews.scala

```diff
@@ -1,21 +1,254 @@
 package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch
 
+import akka.persistence.query.Offset
 import cats.effect.Clock
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.kernel.syntax._
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.ElasticSearchViews._
 import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewCommand._
 import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewEvent._
 import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewRejection._
 import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewState.{Current, Initial}
 import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewValue._
 import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.RemoteContextResolution
+import ch.epfl.bluebrain.nexus.delta.sdk.Projects
+import ch.epfl.bluebrain.nexus.delta.sdk.cache.KeyValueStore
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdSourceParser
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.JsonLdSourceParser.expandIri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment.StringSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
 import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{Project, ProjectRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.Pagination.FromPagination
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.ResultEntry.UnscoredResultEntry
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchResults.UnscoredSearchResults
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, IdSegment, Label}
 import ch.epfl.bluebrain.nexus.delta.sdk.utils.{IOUtils, UUIDF}
+import ch.epfl.bluebrain.nexus.sourcing.processor.EventSourceProcessor.persistenceId
+import ch.epfl.bluebrain.nexus.sourcing.{Aggregate, EventLog}
 import io.circe.Json
-import monix.bio.{IO, UIO}
+import io.circe.syntax._
+import monix.bio.{IO, Task, UIO}
 
-class ElasticSearchViews {}
+final class ElasticSearchViews private (
+    aggregate: ElasticSearchViewAggregate,
+    eventLog: EventLog[Envelope[ElasticSearchViewEvent]],
+    cache: ElasticSearchViewCache,
+    projects: Projects
+)(implicit rcr: RemoteContextResolution, uuidF: UUIDF) {
+
+  def create(
```

Comment:
I would not put this one as view created by other modules should follow provide a meaningful name with a pattern to be distinguishable from user-created ones